lib/grammar/rules/filename.cpp

```diff
@@ -0,0 +1,28 @@
+#include "shell/grammar/rules/filename.hpp"
+#include "shell/grammar/grammar_util.hpp"
+
+namespace shell {
+
+Rule Filename::make() {
+	return Rule {
+	    Filename::options,
+	};
+}
+
+vector<Rule::Option> Filename::options() {
+	return {Rule::Terminal {handler}};
+}
+
+optional<Ast::Node> Filename::handler(TokenProvider &tokens) {
+	auto token = tokens.consumeIf([](Token &token) { return token.getType() == Token::Type::Token; });
+	if (!token.has_value()) {
+		return nullopt;
+	}
+	// Rule 2
+	// shall be subjected to tilde expansion, parameter expansion, command substitution, arithmetic expansion, and quote
+	// removal
+	// TODO: implement the expansions
```

Comment:
I'm a bit too tired to read the standard atm so this comment is a remind me to look at it later.
Is the parser supposed to already apply expansions at this point? I assume no

Reply:
Though since the here-doc rule also mentions expansion, and I assume that should be applied at this point. Maybe that's also the case here.

Reply:
honestly still no clue

Reply:
This is quoted from here:
https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_10

> The expansions specified in [Redirection](https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_07) shall occur

Reply:
Maybe it will become clear once we start implementing the expander, let's just leave a comment for now